src/meta/src/manager/catalog/mod.rs

```diff
@@ -1134,11 +1190,59 @@ impl CatalogManager {
         Ok(version)
     }
 
-    /// This is used for both `CREATE TABLE` and `CREATE MATERIALIZED VIEW`.
+    /// This is used for `CREATE TABLE`.
     pub async fn finish_create_table_procedure(
         &self,
         mut internal_tables: Vec<Table>,
         mut table: Table,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        let mut tables = BTreeMapTransaction::new(&mut database_core.tables);
+        let key = (table.database_id, table.schema_id, table.name.clone());
+        assert!(
+            !tables.contains_key(&table.id)
+                && database_core.in_progress_creation_tracker.contains(&key),
+            "table must be in creating procedure"
+        );
+        database_core.in_progress_creation_tracker.remove(&key);
+        database_core
+            .in_progress_creation_streaming_job
+            .remove(&table.id);
+
+        table.stream_job_status = PbStreamJobStatus::Created.into();
+        tables.insert(table.id, table.clone());
+        for table in &mut internal_tables {
+            table.stream_job_status = PbStreamJobStatus::Created.into();
+            tables.insert(table.id, table.clone());
+        }
+        commit_meta!(self, tables)?;
+
+        tracing::debug!(id = ?table.id, "notifying frontend");
+        let version = self
```

Comment:
Ditto

Reply:
As above, prefer to decouple the functionality.

Reply:
I'm okay with both. Anyway, the code of v1 will eventually be deprecated. 🥵